tests/unit/models/post-test.js

```diff
@@ -153,3 +153,57 @@ test('promiseTypes', function(assert) {
   assert.equal(promiseArray.constructor.superclass, ArrayProxy);
   assert.equal(promiseObject.constructor.superclass, ObjectProxy);
 });
+
+test('model action handles InvalidError', function(assert) {
+  assert.expect(6);
+
+  this.server.put('/posts/:id/publish', (request) => {
+    let data = JSON.parse(request.requestBody);
+    assert.deepEqual(data, { myParam: 'My first param' });
+    assert.equal(request.url, '/posts/1/publish');
+
+    return [422, {}, '{"errors": [{"detail": "First error", "source": {"pointer": "data"}}]}'];
+  });
+
+  let done = assert.async();
+  let payload = { myParam: 'My first param' };
+  let store = this.store();
+  let model = this.subject();
+
+  model.set('id', 1);
+  assert.equal(store.peekAll('post').get('length'), 1);
+
+  model.publish(payload).then((response) => {
```

Comment:
Sorry, I think I missed something. I don't understand why custom actions should remain stateless. Indeed, this is not how ember-data behaves, correct me if I'm wrong ? In your opinion, how should we handle 422 errors sent by the server ? In ember-data, they are put in `model.get('errors')`. Should we not do the same ?